clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -89,6 +114,131 @@ class AzureResourceManagerClient extends AzureBaseClient {
   }
 
   protected ResourceManagementClient getResourceManagementClient(AzureCredentials creds) {
-    return ResourceManagementService.create(this.buildConfiguration(creds))
+    ResourceManagementService.create(this.buildConfiguration(creds))
   }
+
+  static DeploymentExtended createTemplateDeploymentFromPath(
+    ResourceManagementClient resourceManagementClient,
+    String resourceGroupName,
+    DeploymentMode deploymentMode,
+    String deploymentName,
+    String template,
+    Map<String, String> templateParameters) throws URISyntaxException, IOException, ServiceException {
+
+    DeploymentProperties deploymentProperties = new DeploymentProperties();
+    deploymentProperties.setMode(deploymentMode);
+
+    // set the link to template JSON
+    //TemplateLink templateLink = new TemplateLink(new URI(templateUri));
+    //templateLink.setContentVersion(templateContentVersion);
+    deploymentProperties.setTemplate(template);
+
+    // initialize the parameters for this template
+    if (templateParameters) {
+      Map<String, ParameterValue> parameters = new HashMap<String, ParameterValue>();
+      for (Map.Entry<String, String> entry : templateParameters.entrySet()) {
+        parameters.put(entry.getKey(), new ParameterValue(entry.getValue()));
+      }
+      deploymentProperties.setParameters(new Gson().toJson(parameters));
+    }
+
+    // kick off the deployment
+    Deployment deployment = new Deployment();
+    deployment.setProperties(deploymentProperties);
+
+    return resourceManagementClient
+      .getDeploymentsOperations()
+      .createOrUpdate(resourceGroupName, deploymentName, deployment)
+      .getDeployment();
+  }
+
+  private static class ParameterValue {
+    String value;
+    ParameterValue(String value) {
+      this.value = value;
+    }
+  }
+
+
+  /*
+  // Example of how an Azure Load Balancer deploy template should look like
+
+  private static String loadBalancerTemplateString = "{\n" +
```

Comment:
Not sure the value of this?

If anything, it would look better as a multi-line string:

    private static String loadBalancerTemplateString = """
    {
    ...
    }
    """